bundle/config/mutator/cleanup_targets.go

```diff
@@ -0,0 +1,29 @@
+package mutator
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/bundle"
+	"github.com/databricks/cli/libs/diag"
+)
+
+type cleanupTargets struct {
+	name string
+}
+
+// CleanupTargets cleans up configuration properties before the configuration
+// is reported by the 'bundle summary' command.
```

Comment:
Why is this needed?

Reply:
This is the same thing we discussed in #1387: process_target_mode reads the original target but we still do the cleanup for use in validate/summary.

Reply:
I rather keep this working as is and have target selection remove them. By decoupling the cleanup needs to happen everywhere where we do a JSON dump of the configuration.